packages/server/src/product/product.service.ts

```diff
@@ -47,8 +50,41 @@ export class ProductService {
   }
 
   async findAll(): Promise<Product[]> {
-    const products = await this.productRepository.find();
+    const products = await this.productRepository.find({
+      relations: ['attributes', 'store'],
+    });
 
     return products;
   }
+
+  async linkAttribute({ productId, attributesId }: LinkAttributeToProductDto): Promise<Product> {
+    const product = await this.productRepository.findOne(productId, {
+      relations: ['attributes', 'store'],
+    });
+    const attributes = await this.attributeRepository.findByIds(attributesId);
+
+    if (!product) {
+      const errors = { Store: 'not found' };
+      throw new HttpException({ errors }, 401);
+    }
+
+    attributes.forEach(async attribute => {
+      if (!attribute) {
+        const errors = { Attribute: `${attribute.name} not found` };
+        throw new HttpException({ errors }, 401);
```

Comment:
We can apply the same logic related to the status code here

```suggestion
        throw new HttpException({ errors }, HttpStatus.BAD_REQUEST);
```